File: Supercomparator Springboot API/src/main/java/com/tfg/supercomparator/application/service/hipercor/HipercorService.java
package com.tfg.supercomparator.application.service.hipercor;

import com.tfg.supercomparator.domain.hipercor.HipercorProduct;
import com.tfg.supercomparator.domain.hipercor.HipercorProductHistoryItem;
import com.tfg.supercomparator.domain.hipercor.HipercorProductHistoryItems;
import org.springframework.http.ResponseEntity;

import java.util.List;

public interface HipercorService {
    List<HipercorProduct> findProduct(String product);

    ResponseEntity<HipercorProductHistoryItem> saveProductHistory(HipercorProductHistoryItem hipercorProductHistory);

    ResponseEntity<HipercorProductHistoryItems> saveProductsHistory(HipercorProductHistoryItems hipercorProductsHistory);
}

File: SuperComparator/app/src/main/java/com/tfg/supercomparator/service/HipercorService.kt
package com.tfg.supercomparator.service

import com.tfg.supercomparator.domain.modules.model.hipercor.HipercorProduct
import com.tfg.supercomparator.domain.modules.model.hipercor.toProductHistoryItems
import com.tfg.supercomparator.domain.modules.model.product.ProductHistoryItems
import com.tfg.supercomparator.domain.modules.network.HipercorQuoteService
import com.tfg.supercomparator.domain.modules.network.QuoteRepository
import retrofit2.Response

class HipercorService : SearchProduct<HipercorProduct> {
    override suspend fun findProducts(query: String): List<HipercorProduct> {
        return QuoteRepository
            .getSupercomparatorAPIClient()
            .create(HipercorQuoteService::class.java)
            .getHipercorProduct(query)
    }

    override suspend fun saveProductHistory(product: List<HipercorProduct>): Response<ProductHistoryItems> {
        return QuoteRepository
            .getSupercomparatorAPIClient()
            .create(HipercorQuoteService::class.java)
            .saveHistory(product.toProductHistoryItems())
    }
}